### PR TITLE
Resource Tuner: CI-safe test runner (whitelist) with service+config-aware skips — run.sh + README

### PR DESCRIPTION
**Pull request by smuppand**
Why

- Previous discovery-based runs occasionally picked up irrelevant binaries and skewed coverage. We also needed consistent CI gating that respects the service state and presence of required configs without generating false negatives.

What’s in this PR

- Pinned whitelist execution: runs only the approved /usr/bin/*Tests plus /usr/bin/resource_tuner_tests.

- Early service gate: uses repo helper check_systemd_services(resource-tuner.service); if inactive → overall SKIP (no tests run).

Config-aware SKIPs:

- Parser suites require either common/ or custom/:

- common/: InitConfig.yaml, PropertiesConfig.yaml, ResourcesConfig.yaml, SignalsConfig.yaml

- custom/: InitConfig.yaml, PropertiesConfig.yaml, ResourcesConfig.yaml, SignalsConfig.yaml, TargetConfig.yaml, ExtFeaturesConfig.yaml

- resource_tuner_tests also requires tests/Configs/ResourceSysFsNodes (non-empty).

- If requirements aren’t met for a suite, that suite is SKIP only; others continue.

Strict gating:

- Any FAIL ⇒ overall FAIL

- No FAIL & ≥1 PASS ⇒ overall PASS

- No FAIL & 0 PASS (all skipped) ⇒ overall SKIP

Resilient runner:

- Per-suite logs (.log) and result markers (.res)

- Coverage reports (coverage.txt, missing_bins.txt) and counts

- logs/resource-tuner-latest symlink to newest run

- dmesg snapshot captured at start; optional warn-only scanners if toolchain provides them

- Concurrency lock via flock (or dir fallback)

- Optional per-suite timeouts via run_with_timeout if present (ThreadPool/RateLimiter 1800s; resource_tuner_tests 2400s; default 1200s)

Operator-friendly CLI:

- --all (default), --bin <name|path>, --list, --timeout <secs>

- Environment overrides:

- SERVICE_NAME, RT_CONFIG_DIR

- RT_REQUIRE_COMMON_FILES, RT_REQUIRE_CUSTOM_FILES (space-separated lists)

### CODE REVIEW

File: Runner/suites/Performance/resource-tuner/run.sh [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+if [ -z "$INIT_ENV" ]; then
+    echo "[ERROR] Could not find init_env (starting at $SCRIPT_DIR)" >&2
+    exit 1
+fi
+# Only source once (idempotent)
+if [ -z "$__INIT_ENV_LOADED" ]; then
+    # shellcheck disable=SC1090
+    . "$INIT_ENV"
+fi
```
Comment (abbajaj806) [inline Runner/suites/Performance/resource-tuner/run.sh:25]: Once $INIT_ENV is initialized, we should initialize __INIT_ENV_LOADED this as 1?

File: Runner/suites/Performance/resource-tuner/run.sh [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+        echo "total=$total"
+        echo "present=$present"
+        echo "missing=$missing"
+    } > "$LOGDIR/coverage_counts.env"
+    [ $missing -gt 0 ] && log_warn "Whitelist coverage: $present/$total present, $missing missing"
+}
+preflight_bins
+# shellcheck disable=SC1091
+. "$LOGDIR/coverage_counts.env" 2>/dev/null || true
+
```
[the remark this answers is not in the document]
Comment (smuppand) [reply Runner/suites/Performance/resource-tuner/run.sh:257]: you're right even though `preflight_bins` writes the files, guarding the source makes the runner more robust.

[the remark this answers is not in the document]
Comment (smuppand) [reply Runner/suites/Performance/resource-tuner/run.sh:263]: This has been addressed with the most recent changes.

File: Runner/suites/Performance/resource-tuner/run.sh (hunk 1 above)
Comment (smuppand) [reply Runner/suites/Performance/resource-tuner/run.sh:27]: This has been addressed with the most recent changes.
